server/keycloak.js

```diff
@@ -0,0 +1,122 @@
+const querystring = require('querystring');
+const KeyCloakConnect = require('keycloak-connect');
+const axios = require('axios');
+
+const regionMap = {
+  region_fraser: 'Fraser',
+  region_interior: 'Interior',
+  region_northern: 'Northern',
+  region_vancouver_coastal: 'Vancouver Coastal',
+  region_vancouver_island: 'Vancouver Island',
+};
+
+const defaults = {
+  'ssl-required': 'external',
+  'confidential-port': 0,
+  'verify-token-audience': true,
+  'use-resource-role-mappings': true,
+  'policy-enforcer': {},
+};
+
+class Keycloak { // Wrapper class around keycloak-connect
+  constructor() {
+    this.realm = process.env.KEYCLOAK_REALM;
+    this.authUrl = process.env.KEYCLOAK_AUTH_URL;
+    this.clientNameFrontend = process.env.KEYCLOAK_FE_CLIENTID;
+    this.clientNameBackend = process.env.KEYCLOAK_API_CLIENTID;
+    this.clientSecretBackend = process.env.KEYCLOAK_API_SECRET;
+    this.serviceAccountUsername = process.env.KEYCLOAK_SA_USERNAME;
+    this.serviceAccountPassword = process.env.KEYCLOAK_SA_PASSWORD;
+    const config = {
+      ...defaults,
+      realm: this.realm,
+      'auth-server-url': this.authUrl,
+      resource: this.clientNameBackend,
+    };
+    this.keycloakConnect = new KeyCloakConnect({}, config);
+  }
+
+  RealmInfoFrontend() {
+    return {
+      realm: this.realm,
+      url: this.authUrl,
+      clientId: this.clientNameFrontend,
+    };
+  }
+
+  expressMiddleware(...args) { // Default connect middleware for Keycloak connect library
+    return this.keycloakConnect.middleware(...args);
+  }
+
+  allowRolesMiddleware(...roles) { // Connect middleware for limiting roles
+    const allowRoles = () => (token) => {
+      if (roles.length === 0) return true; // Allows any role
```

Comment:
is this temporary?

Reply:
This was to match the way that we were using `keycloak.protect()`. If no roles are provided, it just makes sure that the user is logged in. We could change this and all endpoints that depend on it. Looks like it's only the `/user` API endpoint that uses it.